### PR TITLE
[ENG-587] line model

### PR DESCRIPTION
**Pull request by mgoldmansight**

### CODE REVIEW

File: smsdk/smsdk_entities/line/line.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    # Only the registered utilites would be accessible
+    # to outside world via client.get_data()
+    mod_util = module_utility()
+
+    def __init__(self, session, base_url) -> None:
+        self.session = session
+        self.base_url = base_url
+
+    @mod_util
+    def get_lines(self, *args, **kwargs):
```
Comment (srblum) [inline smsdk/smsdk_entities/line/line.py:37]: What's the use-case for this? Do we want to use the `eb_factory` key in assets for this or `line` (which I believe is derived from `eb_factory`)?

File: smsdk/client.py [hunk 2]
```diff
@@ -347,3 +347,58 @@ def normalize_constraints(self, constraints):
...
+                where.append({"nested": [filter]})
+
+        kwargs["asset_selection"] = asset_selection
+        kwargs["asset_time_offset"] = asset_time_offset
+        kwargs["select"] = fields
+        kwargs["time_selection"] = time_selection
+        kwargs["db_mode"] = 'sql'
+        kwargs["model"] = 'line'
+        kwargs["model_type"] = 'data-table'
+        kwargs["offset"] = 0
```
Comment (srblum) [inline smsdk/client.py:401]: Let's include `limit` as well as `offset`, default limit to `400`, and allow both of them to be overridden by kwargs.

File: smsdk/smsdk_entities/line/line.py (hunk 1 above)
Comment (mgoldmansight) [reply smsdk/smsdk_entities/line/line.py:37]: I'll take a look eb_factory if it has more info I'm happy to use it.  It's to get line info.

File: smsdk/client.py (hunk 2 above)
Comment (mgoldmansight) [reply smsdk/client.py:401]: Sorry do you want both as named parameters or just as setable in kwargs?

Comment (srblum) [reply smsdk/client.py:401]: Probably named parameters would be the most self-documenting?

**Review by srblum — APPROVED** [review]

**Pull request merged**